### ISSUE TITLE
tcmalloc: large alloc while finetune unsupervised model

### ISSUE THREAD

**Issue by h4bUb** [tynguyen/unsupervisedDeepHomographyRAL2018#11]
Hello, i'm trying to finetune unsupervised model using Model trained on Aerial Image Data. I generated training images as you suggest in Generate aerial dataset section, for training i'm using this command:
`python "/home/UDH/code/homography_CNN_real.py" --mode train --lr 1e-4 --loss_type l1_loss --load_model_dir "/home/real_models" --save_model_dir "/home/save_models" --finetune True `

This is the error that i got:
<==================== Loading data ===================>

===> Load model from (if want)  /home/real_models
===> Save model to (if want)  /home/save_models
===> Save visualization to (if want)  /home/report/
===> There are totally 71 test files
===> Train: There are totally 140 training files
args lr: 0.0001 9e-05
===> Decay steps: 58117.5893057
x_t_flat: Tensor("Reshape_1:0", shape=(16384,), dtype=float32, device=/device:GPU:0)
y_t_flat: Tensor("Reshape_2:0", shape=(16384,), dtype=float32, device=/device:GPU:0)
--Shape of A_mat: [64, 8, 8]
--shape of b: [64, 8, 1]
--shape of H_8el Tensor("MatrixSolve:0", shape=(64, 8, 1), dtype=float32, device=/device:GPU:0)
('--Inter- scale_h:', True)
/usr/local/lib/python2.7/dist-packages/tensorflow/python/ops/gradients_impl.py:93: UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory.
  "Converting sparse IndexedSlices to a dense Tensor of unknown shape. "
====> Use loss type:  l1_loss
--Shape of A_mat: [64, 8, 8]
--shape of b: [64, 8, 1]
--shape of H_8el Tensor("MatrixSolve_1:0", shape=(64, 8, 1), dtype=float32, device=/device:GPU:1)
('--Inter- scale_h:', True)
====> Use loss type:  l1_loss
2018-12-07 11:55:48.830193: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.1 instructions, but these are available on your machine and could speed up CPU computations.
2018-12-07 11:55:48.830267: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.2 instructions, but these are available on your machine and could speed up CPU computations.
2018-12-07 11:55:48.830296: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX instructions, but these are available on your machine and could speed up CPU computations.
2018-12-07 11:55:48.830314: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX2 instructions, but these are available on your machine and could speed up CPU computations.
2018-12-07 11:55:48.830333: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use FMA instructions, but these are available on your machine and could speed up CPU computations.
===> Start step: 0
tcmalloc: large alloc 2684600320 bytes == 0x557650d7c000 @  0x7f031dcac1e7 0x7f0311d364ef 0x7f03131f3869 0x7f031320027d 0x7f031322c040 0x7f031322c4ae 0x7f031322d75f 0x7f031340e81c 0x7f03133df108 0x7f03133df8ca 0x7f0313723a92 0x7f0313722d22 0x7f031c7c08f0 0x7f031d6706db 0x7f031d9a988f
tcmalloc: large alloc 2684600320 bytes == 0x557650d7c000 @  0x7f031dcac1e7 0x7f0311d364ef 0x7f03131f3869 0x7f031320027d 0x7f031322c040 0x7f031322c4ae 0x7f031322d75f 0x7f031340e81c 0x7f03133df108 0x7f03133ce630 0x7f0313723a92 0x7f0313722d22 0x7f031c7c08f0 0x7f031d6706db 0x7f031d9a988f
tcmalloc: large alloc 2684600320 bytes == 0x557651c18000 @  0x7f031dcac1e7 0x7f0311d364ef 0x7f03131f3869 0x7f031320027d 0x7f031322c040 0x7f031322c4ae 0x7f031322d75f 0x7f031340e81c 0x7f03133df108 0x7f03133ce630 0x7f0313723a92 0x7f0313722d22 0x7f031c7c08f0 0x7f031d6706db 0x7f031d9a988f
tcmalloc: large alloc 2684600320 bytes == 0x5576f1c54000 @  0x7f031dcac1e7 0x7f0311d364ef 0x7f03131f3869 0x7f031320027d 0x7f031322c040 0x7f031322c4ae 0x7f031322d75f 0x7f031340e81c 0x7f03133df108 0x7f03133ce630 0x7f0313723a92 0x7f0313722d22 0x7f031c7c08f0 0x7f031d6706db 0x7f031d9a988f
tcmalloc: large alloc 2684600320 bytes == 0x557651c18000 @  0x7f031dcac1e7 0x7f0311d364ef 0x7f03131f3869 0x7f031320027d 0x7f031322c040 0x7f031322c4ae 0x7f031322d75f 0x7f031340e81c 0x7f03133df108 0x7f03133df8ca 0x7f0313723a92 0x7f0313722d22 0x7f031c7c08f0 0x7f031d6706db 0x7f031d9a988f
tcmalloc: large alloc 2684600320 bytes == 0x557679c46000 @  0x7f031dcac1e7 0x7f0311d364ef 0x7f03131f3869 0x7f031320027d 0x7f031322c040 0x7f031322c4ae 0x7f031322d75f 0x7f031340e81c 0x7f03133df108 0x7f03133ce630 0x7f0313723a92 0x7f0313722d22 0x7f031c7c08f0 0x7f031d6706db 0x7f031d9a988f
tcmalloc: large alloc 2684600320 bytes == 0x557651c18000 @  0x7f031dcac1e7 0x7f0311d364ef 0x7f03131f3869 0x7f031320027d 0x7f031322c040 0x7f031322c4ae 0x7f031322d75f 0x7f031340e81c 0x7f03133df108 0x7f03133df8ca 0x7f0313723a92 0x7f0313722d22 0x7f031c7c08f0 0x7f031d6706db 0x7f031d9a988f
tcmalloc: large alloc 2684600320 bytes == 0x557651c18000 @  0x7f031dcac1e7 0x7f0311d364ef 0x7f03131f3869 0x7f031320027d 0x7f031322c040 0x7f031322c4ae 0x7f031322d75f 0x7f031340e81c 0x7f03133df108 0x7f03133ce630 0x7f0313723a92 0x7f0313722d22 0x7f031c7c08f0 0x7f031d6706db 0x7f031d9a988f

after that script freezes and nothing happens

I run all scripts in google colaboratory on python 2.7 and gpu support, tensorflow version is 1.2.1

Also i found that patches in gen_real_data.py are used only in visualization, but according to papper NN needs this patches, are they generating somewhere else?
`# grab image patches`
`I1   = I_gray[y:y + patch_size, x:x + patch_size]`
`I2   = I_prime_gray[y:y + patch_size, x:x + patch_size]`
